sources/packages/backend/apps/api/src/route-controllers/application-change-request/_tests_/e2e/application-change-request.aest.assessApplicationChangeRequest.e2e-spec.ts

```diff
@@ -0,0 +1,482 @@
+    app = nestApplication;
+    appDataSource = dataSource;
+    zbClient = app.get(ZeebeGrpcClient);
+    db = createE2EDataSources(dataSource);
+    const auditUser = await getAESTUser(
+      dataSource,
+      AESTGroups.BusinessAdministrators,
+    );
+    ministryUser = { id: auditUser.id } as User;
+  });
+
+  beforeEach(async () => {
+    MockDate.reset();
+  });
+
+  it("Should approve a change request and copy the offering and appeal when the application change request is waiting for approval.", async () => {
+    // Arrange
+    // Application to be replaced by the change request.
+    const applicationToBeReplaced = await saveFakeApplication(
+      appDataSource,
+      undefined,
+      {
+        initialValues: {
+          applicationStatus: ApplicationStatus.Completed,
+          applicationEditStatus: ApplicationEditStatus.Original,
+        },
+      },
+    );
+    // Create a student appeal and have it associated with the "applicationToBeReplaced".
+    const appeal = createFakeStudentAppeal({
+      application: applicationToBeReplaced,
+    });
+    applicationToBeReplaced.currentAssessment.studentAppeal = appeal;
+    await db.application.save(applicationToBeReplaced);
+    // Change request to be approved replacing the "applicationToBeReplaced".
+    const changeRequest = await saveFakeApplication(
+      appDataSource,
+      { precedingApplication: applicationToBeReplaced },
+      {
+        initialValues: {
+          applicationStatus: ApplicationStatus.Edited,
+          applicationEditStatus: ApplicationEditStatus.ChangePendingApproval,
+        },
+      },
+    );
+    // Offering to perform the asserts. Expected to be copied.
+    const expectedOffering = {
+      id: applicationToBeReplaced.currentAssessment.offering.id,
+    };
+    // Appeal to perform the asserts. Expected to be copied.
+    const expectedAppeal = { id: appeal.id };
+    const payload = getPayload(ApplicationEditStatus.ChangedWithApproval);
+    const endpoint = `/aest/application-change-request/${changeRequest.id}`;
+    const token = await getAESTToken(AESTGroups.BusinessAdministrators);
+    const now = new Date();
+    MockDate.set(now);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .patch(endpoint)
+      .send(payload)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK);
+    // Validate workflow message.
+    expect(zbClient.publishMessage).toHaveBeenCalledWith(
+      expect.objectContaining({
+        correlationKey: changeRequest.id.toString(),
+        name: "application-change-request-status-message",
+        variables: {
+          applicationEditStatus: ApplicationEditStatus.ChangedWithApproval,
+        },
+      }),
+    );
+    // Validate database changes.
+    const applicationChangeRequest = await db.application.findOne({
+      select: {
+        id: true,
+        applicationStatus: true,
+        applicationStatusUpdatedOn: true,
```

Comment:
applicationEditStatusUpdatedOn and applicationEditStatusUpdatedBy can be validated.

Reply:
They are not participating in the DB changes. Is the idea to validate if they were not changed by mistake?

Reply:
Missed! Those ones were pretty much the core of it.